linera-rpc/src/grpc/client.rs

```diff
@@ -364,6 +364,15 @@ impl ValidatorNode for GrpcClient {
             missing_hashes = missing_hashes[received.len()..].to_vec();
             certs_collected.append(&mut received);
         }
+        if !missing_hashes.is_empty() {
+            return Err(NodeError::MissingCertificates {
+                missing: missing_hashes
+                    .into_iter()
+                    .map(|hash| hash.to_string())
+                    .collect::<Vec<_>>()
+                    .join(", "),
+            });
+        }
```

Comment:
I'd receive `download_certificate` here, too, i.e. also from `rpc.proto`?

Reply:
Remove, not receive. Sorry, long day. :sweat_smile: 

I mean, we still have a gRPC message `DownloadCertificate` (singular), don't we?

Reply:
I'll take a look at that (once I deal with the error at hand) but I thought we'd have to deal with the possiblity of having more than one element in the result on the caller's side. Here it's hidden and if caller requests 1 certificate, the result is `ConfirmedBlockCertificate`, not a `Vec<ConfirmedBlockCertificate>`. I'd prefer to avoid too many `unwrap()` – here it's hidden.

Reply:
Good point, you're right!